packages/react/test/specs/components/Accordion/Accordion-test.tsx

```diff
@@ -0,0 +1,227 @@
+        key: 'ArrowUp',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(0)
+
+      getTitleButtonAtIndex(wrapper, 0).simulate('keydown', {
+        keyCode: keyboardKey.ArrowDown,
+        key: 'ArrowDown',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(1)
+    })
+
+    it('is changed by arrow key navigation in a circular way', () => {
+      const wrapper = mountWithProvider(<Accordion panels={panels} />)
+      const accordion = wrapper.find(Accordion)
+      getTitleButtonAtIndex(wrapper, 0).simulate('click')
+      getTitleButtonAtIndex(wrapper, 0).simulate('keydown', {
+        keyCode: keyboardKey.ArrowUp,
+        key: 'ArrowUp',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(panels.length - 1)
+
+      getTitleButtonAtIndex(wrapper, panels.length - 1).simulate('keydown', {
+        keyCode: keyboardKey.ArrowDown,
+        key: 'ArrowDown',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(0)
+    })
+
+    it('is changed to `0` at Home keydown', () => {
+      const wrapper = mountWithProvider(<Accordion panels={panels} />)
+      const accordion = wrapper.find(Accordion)
+      getTitleButtonAtIndex(wrapper, 2).simulate('click')
+      getTitleButtonAtIndex(wrapper, 2).simulate('keydown', {
+        keyCode: keyboardKey.Home,
+        key: 'Home',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(0)
+    })
+
+    it('is changed to last index at End keydown', () => {
+      const wrapper = mountWithProvider(<Accordion panels={panels} />)
+      const accordion = wrapper.find(Accordion)
+      getTitleButtonAtIndex(wrapper, 0).simulate('click')
+      getTitleButtonAtIndex(wrapper, 0).simulate('keydown', {
+        keyCode: keyboardKey.End,
+        key: 'End',
+      })
+      expect(accordion.state('focusedIndex')).toEqual(panels.length - 1)
+    })
+
+    it('focuses the button element when is changed via focus handler', () => {
+      const wrapper = mountWithProvider(<Accordion panels={panels} />)
+      const title = getTitleButtonAtIndex(wrapper, 1)
+      title.simulate('click')
+      title.simulate('keydown', { keyCode: keyboardKey.ArrowUp, key: 'ArrowUp' })
+      expect(document.activeElement).toEqual(getTitleButtonAtIndex(wrapper, 0).getDOMNode())
+    })
+  })
+
+  describe('panels', () => {
+    it('when clicked call onClick and onTitleClick if provided by the user', () => {
+      const onTitleClick = jest.fn()
+      const onClick = jest.fn()
+      const panels = [
+        {
+          key: 'one',
+          title: { key: 'oneTitle', content: 'One', onClick },
+          content: '2 3 4',
+        },
+      ]
+      const wrapper = mountWithProvider(<Accordion panels={panels} onTitleClick={onTitleClick} />)
+      getTitleButtonAtIndex(wrapper, 0).simulate('click')
+
+      expect(onTitleClick).toBeCalledTimes(1)
+      expect(onClick).toBeCalledTimes(1)
+    })
+
+    it('when focused call onFocus if provided by the user', () => {
```

Comment:
Isn't this tested already by the conformance's handlers events transparently? Let's just check, before removing this test.

Reply:
we agreed on the fix